tests/testthat/test-facet-.r

```diff
@@ -90,6 +90,18 @@ test_that("facets split up the data", {
   expect_equal(d1$PANEL, factor(1:3))
 })
 
+test_that("facets handle empty layers", {
+  p <- ggplot() + geom_point(aes(x, y), df) + geom_line()
```

Comment:
The way `df` is being reused in this file is a bit strange. Initially it's defined outside of `test_that()` and reused many times, but later on it's overwritten inside various `test_that()` alls. It makes it difficult to figure out what `df` actually is at any point in the code. Maybe refactor through a combination of moving the creation of `df` into `test_that()` calls and combining some tests into one  `test_that()` call so you don't have to recreate `df` over and over?

Reply:
Makes sense. I'll try it.